File: Webfuel.Api/Api/Common/FileStorageApi.cs
using MediatR;
using Webfuel.Common;
using Webfuel.Domain;
using static Webfuel.App.PingApi;

namespace Webfuel.App
{
    [ApiService]
    public static class FileStorageApi
    {
        public static void RegisterEndpoints(IEndpointRouteBuilder app)
        {
            app.MapGet("api/file-storage/files/{fileStorageGroupId:guid}", ListFiles)
                 .RequireIdentity();

            app.MapDelete("api/file-storage/files/{fileStorageEntryId:guid}", ListFiles)
                .RequireIdentity();
        }

        public static async Task<List<FileStorageEntry>> ListFiles(Guid fileStorageGroupId, IFileStorageService fileStorageService)
        {
            return await fileStorageService.ListFiles(fileStorageGroupId);
        }

        public static async Task DeleteFile(Guid fileStorageEntryId, IFileStorageService fileStorageService)
        {
            await fileStorageService.DeleteFile(fileStorageEntryId);
        }
    }
}

File: Webfuel.App/ClientApp/src/app/api/file-storage.api.ts
import { EventEmitter, Injectable, inject } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { ApiService, ApiOptions } from '../core/api.service';
import { ActivatedRouteSnapshot, ResolveFn, RouterStateSnapshot } from '@angular/router';
import { IDataSource } from 'shared/common/data-source';
import { FileStorageEntry, StringResult } from './api.types';

@Injectable()
export class FileStorageApi {
    constructor(private apiService: ApiService) { }
    
    public listFiles (params: { fileStorageGroupId: string }, options?: ApiOptions): Observable<Array<FileStorageEntry>> {
        return this.apiService.request<undefined, Array<FileStorageEntry>>("GET", "api/file-storage/files/" + params.fileStorageGroupId + "", undefined, options);
    }
    
    public generateFileSasUri (params: { fileStorageEntryId: string }, options?: ApiOptions): Observable<StringResult> {
        return this.apiService.request<undefined, StringResult>("GET", "api/file-storage/sas-uri/" + params.fileStorageEntryId + "", undefined, options);
    }
    
    public deleteFiles (params: { fileStorageEntryId: string }, options?: ApiOptions): Observable<any> {
        return this.apiService.request<undefined, any>("DELETE", "api/file-storage/files/" + params.fileStorageEntryId + "", undefined, options);
    }
}

